CRM/Contribute/BAO/Contribution.php

```diff
@@ -3387,6 +3387,7 @@ public static function recordFinancialAccounts(&$params, CRM_Contribute_DAO_Cont
     ) {
       $eventID = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_Participant', $entityId, 'event_id');
       $feeLevel[] = str_replace('', '', $params['prevContribution']->amount_level);
+      $feeLevel = (string) current($feeLevel);
```

Comment:
These two lines seem like a non-sequitur.  First it adds an item to an array, then overwrites the array with that item? Why not just
```suggestion
      $feeLevel = str_replace('', '', $params['prevContribution']->amount_level);
```

Reply:
Yeah I think I focussed on moving the code to the right place - but sometimes when you do that the next step becomes clear!